src/pudl/analysis/record_linkage/embed_dataframe.py

```diff
@@ -150,12 +153,13 @@ def as_transformer(self):
 
 
 def _apply_cleaning_func(df, function_key: str = None):
+    category_cols = df.select_dtypes(include="category").columns
+    df[category_cols] = df[category_cols].astype("str")
```

Comment:
I'm not sure why, but I was getting errors when cleaning the `category` type columns because converting nulls to the empty string adds another category value that's not set in `categories`. Maybe I should reset the categories themselves instead of converting all category columns to strings?

Reply:
The plant and construction types are both being converted to strings in the ferc-ferc model before vectorizing, so I think this is pretty much equivalent to how that works